### PR TITLE
CDAP-2464 Documentation to highlight different CDAP modes

### PR DESCRIPTION
**Pull request by awholegunch**
**Summary**
- Add text and links about different CDAP modes available to a ["Modes and Components"](http://builds.cask.co/artifact/CDAP-DOB52/shared/build-8/Docs-HTML/3.3.0-SNAPSHOT/en/developers-manual/overview/modes.html) page.
- Moves Table of Contents that was on the index page to [a new page](http://builds.cask.co/artifact/CDAP-DOB52/shared/build-8/Docs-HTML/3.3.0-SNAPSHOT/en/table-of-contents.html).
- Adds link to that overall Table of Contents to the left sidebar, at the end of the list.
- Adds a tag to create a link-only table of contents, and fixes other issues found along the way.
- Fixes a problem with the highlighting of the left sidebar current entry.

**Pages of Interest**
- [Overview](http://builds.cask.co/artifact/CDAP-DOB52/shared/build-9/Docs-HTML/3.3.0-SNAPSHOT/en/index.html)
- [Table of Contents](http://builds.cask.co/artifact/CDAP-DOB52/shared/build-9/Docs-HTML/3.3.0-SNAPSHOT/en/table-of-contents.html)
- [Developers’ Manual Overview](http://builds.cask.co/artifact/CDAP-DOB52/shared/build-9/Docs-HTML/3.3.0-SNAPSHOT/en/developers-manual/overview/index.html)
- [Modes and Components](http://builds.cask.co/artifact/CDAP-DOB52/shared/build-9/Docs-HTML/3.3.0-SNAPSHOT/en/developers-manual/overview/modes.html)

Building [Quick Build](http://builds.cask.co/browse/CDAP-DOB52-9). *(Updated for Build 9.)*
Incorporates material from https://github.com/caskdata/cdap/pull/4604
Fix for https://issues.cask.co/browse/CDAP-2464

### CODE REVIEW

**Comment by awholegunch** [comment]
Needs revisions.

**Comment by awholegunch** [comment]
Ready for review...

File: cdap-docs/_common/_source/index.rst [hunk 1]
```diff
@@ -45,117 +44,78 @@ It includes the CDAP programming APIs and client interfaces, with instructions
 on the installation, monitoring and diagnosing fully distributed CDAP in a Hadoop cluster.
 
 
-.. |introduction| replace:: **Introduction to CDAP:**
```
Comment (jwang47) [inline cdap-docs/_common/_source/index.rst]: Why was this section moved out?

Comment (awholegunch) [inline cdap-docs/_common/_source/index.rst]: Because Sree requested it.

**Comment by anwar6953** [comment]
Closing; this documentation PR will have to be reopened against release/3.3 branch

**Pull request closed without merging**